numpy/typing/tests/test_typing.py

```diff
@@ -25,6 +26,15 @@
 CACHE_DIR = os.path.join(DATA_DIR, ".mypy_cache")
 
 
+@pytest.mark.slow
+@pytest.mark.skipif(NO_MYPY, reason="Mypy is not installed")
```

Comment:
The `slow` and `skipif` markers are copied from the actual tests further down below.

While not strictly necessary, this does mean that the cache is _only_ cleared when one is actually running the typing tests (which I feel is a desirable property).

Reply:
I fail to see why `pytest` would care, the decorated function is not a test and I don't see why it would ever get executed. Is it called anywhere?

Reply:
The `autouse=True` argument of `@pytest.fixture` ensures that the function will be automatically executed _before_ running any of the actual tests. This also means it is insensitive towards, for example, test randomization.

Secondly, a bit of trial and error shows that fixtures it does respect the `@pytest.mark` decorators.

Reply:
Thanks for the explanation.